cmd/probe/tcp_probe_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+)
+
+func TestTCPProbe(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), false)
+
+		if err := p.Run(t.Context()); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("conn should fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionSucceeded) {
+			t.Fatalf("expecting error %v, got %v", errConnectionSucceeded, err)
+		}
+	})
+
+	t.Run("conn should not fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		l.Close()
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionFailed) {
+			t.Fatalf("expecting error %v, got %v", errConnectionFailed, err)
+		}
+	})
+
+	t.Run("context aware", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
```

Comment:
If I understand the intent of this test correctly, this should resolve the issue -- expecting failure here is causing the error to be swallowed up.
```suggestion
		p := NewTCPProbe(l.Addr().String(), false)
```

Reply:
Yes, correct! See my [comment above](https://github.com/grafana/nethax/pull/45#discussion_r2081578664), I just forgot to push the changes.